### ISSUE TITLE
bug(form-field): Extra navigation stop on label with VoiceOver and ChromeVox

### ISSUE THREAD

**Issue by jelbourn** [angular/components#22940]
This can be reproduced on the docs site:
https://material.angular.io/components/select/examples
https://material.angular.io/components/autocomplete/examples

When navigating the page with either VoiceOver or ChromeVox, you'll land on the label first before landing on the control itself (which also reads the label).

**Comment by Splaktar** [comment]
This seems like fairly expected behavior. Using the left/right arrows or VO-Left/Right arrows, Voiceover stops on headers, labels, text elements, and interactive elements.

If I put `aria-hidden="true"` on the `label`, then Voiceover navigation completely skips the `mat-select` in addition to the `<label>` .

Can you confirm that the desired outcome here is that 
- the text in the label would be read when the `mat-select` is navigated to
- the `<label>` that contains the `<mat-label>` would not be accessible via the left/right arrows or VO-Left/Right arrows in Voiceover

**Comment by jelbourn** [comment]
The control should be a single navigation stop that announces itself as a combobox, even if it does read the label twice on that navigation stop. [This page](http://www.google.com/url?sa=D&q=https%3A%2F%2Fwww.google.com%2Ftravel%2Fflights%2Fcheckout%3Ftfs%3DCBwQAhqWAmoMCAISCC9tLzBkNmxwEgoyMDIxLTA2LTMwcgwIAhIIL20vMGg3aDYidgoDU0ZPEgoyMDIxLTA2LTMwGgNZVlIqAkFDMgQ4ODM5OlQIARIBQRoIQU5BNEExVEciAkFDKgQ4ODM5MhkyMDIxLTA2LTMwVDA5OjIwOjAwLTA3OjAwOhkyMDIxLTA2LTMwVDExOjQwOjAwLTA3OjAwQgNDUjkidAoDWVZSEgoyMDIxLTA2LTMwGgNZWVoqAkFDMgMxMTY6UwgBEgFBGghBTkE0QTFURyICQUMqAzExNjIZMjAyMS0wNi0zMFQxMzozMDowMC0wNzowMDoZMjAyMS0wNi0zMFQyMDo0ODowMC0wNDowMEIDNzg5GowDagwIAhIIL20vMGg3aDYSCjIwMjEtMDctMDRyDAgCEggvbS8wZDZscCJ0CgNZWVoSCjIwMjEtMDctMDQaA1lXRyoCQUMyAzI1OTpTCAESAUEaCEFOQTRBMVRHIgJBQyoDMjU5MhkyMDIxLTA3LTA0VDA4OjU1OjAwLTA0OjAwOhkyMDIxLTA3LTA0VDEwOjMwOjAwLTA1OjAwQgMyMjMidAoDWVdHEgoyMDIxLTA3LTA0GgNZVlIqAkFDMgMyOTc6UwgBEgFBGghBTkE0QTFURyICQUMqAzI5NzIZMjAyMS0wNy0wNFQxMTowMDowMC0wNTowMDoZMjAyMS0wNy0wNFQxMjowMDowMC0wNzowMEIDMzIwInYKA1lWUhIKMjAyMS0wNy0wNBoDU0ZPKgJBQzIEODg0NDpUCAESAUEaCEFOQTRBMVRHIgJBQyoEODg0NDIZMjAyMS0wNy0wNFQxMzozNTowMC0wNzowMDoZMjAyMS0wNy0wNFQxNjowNTowMC0wNzowMEIDQ1I5cAGCAQsI____________AUABSAGYAQGiAc8BCgJBQxIMCgoKA1VTRBCiBBgAGg4KAkFDEghTVEFOREFSREJrQUFuQzlSc1JwWUh0Y2lTbXZ2NlYwMkZqZTVlRjQ3SHh4T3lfeVR0RnV4VWFsZGFLaU1xMHV5STlaN1R3UWVMVVZISzlESXJfM1NkM2tUSFpuMFZyZnpVM1RfQU1tLVJlbDZTVTZiZFBWM1EiNEh2UHVLQmJURlRlTUFBTUtBUUJELS0tLS0tLS0tcGZyeDI4QUFBQUFHREhyRndBWXBXQUFKCENBSVFBUT09%26tfu%3DEgYIARAAGAA%26gclid%3DAAnC9RsRpYHtciSmvv6V02Fje5eF47HxxOy_yTtFuxUaldaKiMq0uyI9Z7TwQeLUVHK9DIr_3Sd3kTHZn0VrfzU3T_AMm-Rel6SU6bdPV3Q%26gclsrc%3Dgf) on Google's travel site has some MDC-based (but not Angular) components that show the behavior.

**Comment by Splaktar** [comment]
This may have something to do with our DOM structure.

## Mat
![Screen Shot 2021-06-21 at 13 51 05](https://user-images.githubusercontent.com/3506071/122806063-dfb4e100-d297-11eb-88fc-b06d5f67e884.png)


## Mat-MDC
![Screen Shot 2021-06-21 at 13 50 40](https://user-images.githubusercontent.com/3506071/122806055-dd528700-d297-11eb-85a9-dfc58ec4bd71.png)


### MDC
![Screen Shot 2021-06-21 at 13 47 58](https://user-images.githubusercontent.com/3506071/122806035-d7f53c80-d297-11eb-8704-b0e8bc6dcea6.png)

## Summary

Mat and Mat-MDC do not wrap the `input` in the `label`, but MDC does have the `input` wrapped by the `label` element.

Additionally, MDC uses `aria-labelledby` to point to the input's "label" (which is actually a `span` nested in the `label`) and `aria-describedby` to point to the hint/validation container.

**Comment by Splaktar** [comment]
Related comments on why we chose our current DOM structure instead of using the approach that MDC uses:
https://github.com/angular/components/blob/23a0248bc882974258a3b53bddb3c9ca077eeb2b/src/material-experimental/mdc-form-field/form-field.html#L3-L15

**Comment by Splaktar** [comment]
> [This page](http://www.google.com/url?sa=D&q=https%3A%2F%2Fwww.google.com%2Ftravel%2Fflights%2Fcheckout%3Ftfs%3DCBwQAhqWAmoMCAISCC9tLzBkNmxwEgoyMDIxLTA2LTMwcgwIAhIIL20vMGg3aDYidgoDU0ZPEgoyMDIxLTA2LTMwGgNZVlIqAkFDMgQ4ODM5OlQIARIBQRoIQU5BNEExVEciAkFDKgQ4ODM5MhkyMDIxLTA2LTMwVDA5OjIwOjAwLTA3OjAwOhkyMDIxLTA2LTMwVDExOjQwOjAwLTA3OjAwQgNDUjkidAoDWVZSEgoyMDIxLTA2LTMwGgNZWVoqAkFDMgMxMTY6UwgBEgFBGghBTkE0QTFURyICQUMqAzExNjIZMjAyMS0wNi0zMFQxMzozMDowMC0wNzowMDoZMjAyMS0wNi0zMFQyMDo0ODowMC0wNDowMEIDNzg5GowDagwIAhIIL20vMGg3aDYSCjIwMjEtMDctMDRyDAgCEggvbS8wZDZscCJ0CgNZWVoSCjIwMjEtMDctMDQaA1lXRyoCQUMyAzI1OTpTCAESAUEaCEFOQTRBMVRHIgJBQyoDMjU5MhkyMDIxLTA3LTA0VDA4OjU1OjAwLTA0OjAwOhkyMDIxLTA3LTA0VDEwOjMwOjAwLTA1OjAwQgMyMjMidAoDWVdHEgoyMDIxLTA3LTA0GgNZVlIqAkFDMgMyOTc6UwgBEgFBGghBTkE0QTFURyICQUMqAzI5NzIZMjAyMS0wNy0wNFQxMTowMDowMC0wNTowMDoZMjAyMS0wNy0wNFQxMjowMDowMC0wNzowMEIDMzIwInYKA1lWUhIKMjAyMS0wNy0wNBoDU0ZPKgJBQzIEODg0NDpUCAESAUEaCEFOQTRBMVRHIgJBQyoEODg0NDIZMjAyMS0wNy0wNFQxMzozNTowMC0wNzowMDoZMjAyMS0wNy0wNFQxNjowNTowMC0wNzowMEIDQ1I5cAGCAQsI____________AUABSAGYAQGiAc8BCgJBQxIMCgoKA1VTRBCiBBgAGg4KAkFDEghTVEFOREFSREJrQUFuQzlSc1JwWUh0Y2lTbXZ2NlYwMkZqZTVlRjQ3SHh4T3lfeVR0RnV4VWFsZGFLaU1xMHV5STlaN1R3UWVMVVZISzlESXJfM1NkM2tUSFpuMFZyZnpVM1RfQU1tLVJlbDZTVTZiZFBWM1EiNEh2UHVLQmJURlRlTUFBTUtBUUJELS0tLS0tLS0tcGZyeDI4QUFBQUFHREhyRndBWXBXQUFKCENBSVFBUT09%26tfu%3DEgYIARAAGAA%26gclid%3DAAnC9RsRpYHtciSmvv6V02Fje5eF47HxxOy_yTtFuxUaldaKiMq0uyI9Z7TwQeLUVHK9DIr_3Sd3kTHZn0VrfzU3T_AMm-Rel6SU6bdPV3Q%26gclsrc%3Dgf) on Google's travel site has some MDC-based (but not Angular) components that show the behavior.

This page doesn't seem to work with VoiceOver navigation controls for me. Neither left/right arrow or VO+Left/Right arrow move the selection, they just repeat and get stuck on the same element. This problematic behavior isn't reproducible on our docs site or dev-app.

**Comment by Splaktar** [comment]
On https://material.io/components/text-fields, when using VoiceOver navigation, it lands on the `input` first and then the `label`. So there are still two navigation stops with the pure MDC components, they just invest the order (`mat-form-field` lands on the `label` first and then the `input`).

**Comment by Splaktar** [comment]
On ChromeVox:
- mat-mdc only has single tab stops, but there are two navigation stops for most cases with `input`s and `textarea`s
- MDC has only single tabs stops and single navigation stops

**Comment by Splaktar** [comment]
The legacy mat-form-fields don't have a navigation stop on the asterisk, but they do have stops on the label, input, and hint.

**Comment by Splaktar** [comment]
So I guess that I can't compare against the MDC components on https://material.io/components or https://material-components.github.io/material-components-web-catalog since it appears that those sites don't get updated and are using MDC from ~2016. I've asked for where I can find the catalog of components published with the latest release, but it appears that this may not be available.

**Comment by Splaktar** [comment]
As for the extra navigation stop on the asterisk, this is related to the following style from MDC
```sass
.mdc-floating-label--required::after {
    margin-left: 1px;
    margin-right: 0px;
    content: "*";
}
```
This might be a bug in ChromeVox as this isn't reproducible in VoiceOver.

**Comment by Splaktar** [comment]
From the MDC team

> For linear SR navigation with all elements present, the text field should go through:
> - label
> - leading icon (only if interactive)
> - prefix
> - input
> - suffix
> - trailing icon (only if interactive)
> - helper text
> 
> The character counter is hidden from the screen reader.

This is based upon the latest release of MDC, which isn't available to demo/verify anywhere atm.

That said, it does appear that having a separate navigation stop for the label and input is expected. @jelbourn assuming the MDC team has run this all through the a11y team, should we align with them and keep the SR navigation stops as is for label and input?

**Comment by jelbourn** [comment]
Yeah, I think that's enough to satisfy this in the name of consistency, though I personally think that the extra stop for the label is a poor user experience.

**Comment by Splaktar** [comment]
Jeremy is going to check with the a11y team to determine desired outcome and next steps.

**Comment by jelbourn** [comment]
I can confirm that the extra navigation stop on the label is expected and intended.
